stonesoup/types/array.py

```diff
@@ -2,35 +2,72 @@
 import numpy as np
 
 
-class StateVector(np.ndarray):
-    """State vector wrapper for :class:`numpy.ndarray`
+class Matrix(np.ndarray):
+    """Matrix wrapper for :class:`numpy.ndarray`
 
-    This class returns a view to a :class:`numpy.ndarray`, but ensures that
-    its initialised at a *Nx1* vector. It's called same as to
-    :func:`numpy.asarray`.
+    This class returns a view to a :class:`numpy.ndarray` It's called same as
+    to :func:`numpy.asarray`.
     """
 
     def __new__(cls, *args, **kwargs):
         array = np.asarray(*args, **kwargs)
-        if not (array.ndim == 2 and array.shape[1] == 1):
-            raise ValueError(
-                "state vector shape should be Nx1 dimensions: got {}".format(
-                    array.shape))
         return array.view(cls)
 
     def __array_wrap__(self, array):
-        return np.asarray(array)
+        return Matrix._cast(np.asarray(array))
+
+    @classmethod
+    def _cast(cls, val):
+        # This tries to cast the result as either a StateVector or
+        # Matrix type if applicable.
+        if isinstance(val, np.ndarray):
+            if val.ndim == 2:
+                if val.shape[1] == 1:
+                    return val.view(StateVector)
+                else:
+                    return val.view(Matrix)
+            else:
+                return val.view(Matrix)
+        else:
+            return val
```

Comment:
I think this could be a little simpler:
```diff
diff --git a/stonesoup/types/array.py b/stonesoup/types/array.py
index 80ee741..c535017 100644
--- a/stonesoup/types/array.py
+++ b/stonesoup/types/array.py
@@ -21,11 +21,8 @@ class Matrix(np.ndarray):
         # This tries to cast the result as either a StateVector or
         # Matrix type if applicable.
         if isinstance(val, np.ndarray):
-            if val.ndim == 2:
-                if val.shape[1] == 1:
-                    return val.view(StateVector)
-                else:
-                    return val.view(Matrix)
+            if val.ndim == 2 and val.shape[1] == 1:
+                return val.view(StateVector)
             else:
                 return val.view(Matrix)
         else:
```